### PR TITLE
Implement context import/export

### PR DESCRIPTION
**Pull request by sfl-zwang**
### Changes proposed in this pull request:

Added context import/export for the existing project and variant. 

### Status

- [x] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### How to verify this change

Import and export buttons are added to the AI context page with a helper text explaining how import works.

image

The export button opens a dropdown to select the variants to export.

image

Example context file format for import:

```json
{
  "exported_at": "2026-07-24T15:45:55.323890+00:00",
  "projects": [
    {
      "project_description": "test",
      "project_name": "test",
      "variants": [
        {
          "codebase_path": null,
          "environment": "test",
          "other_info": null,
          "risks": null,
          "threat_model": "test",
          "variant_description": null,
          "variant_name": "default"
        },
        {
          "codebase_path": null,
          "environment": null,
          "other_info": null,
          "risks": null,
          "threat_model": "test",
          "variant_description": null,
          "variant_name": "test"
        }
      ]
    },
    {
      "project_description": "null",
      "project_name": "test",
      "variants": [
        {
          "codebase_path": null,
          "environment": null,
          "other_info": null,
          "risks": null,
          "threat_model": "test",
          "variant_description": null,
          "variant_name": "test"
        }
      ]
    }
  ],
  "version": "1.0"
}
```
The importer goes through each entry in the JSON file to validate the context. A successful import requires the following:
1. The project and the variant exist in VulnScout.
2. Required fields are present: `description`, `project_name`, `variant_name` and `threat_model`.

Support for import/export is also added to the CLI commands.

Example commands to import/export: 
```bash
# Export the context for a given project and variant
./vulnscout --project <your-project> --variant <your-variant> --export-context
# Import a context JSON file for a project
./vulnscout --project <your-project> --import-context "path/to/context.json"
```

### Additional notes

1. The import overwrites the existing context entirely when the provided project and variant match the ones in VulnScout.
2. Invalid fields will be ignored.
3. The new changes will be committed to the DB when the entire import is successful.

### Related Issue

*If this PR relates to an issue, please link it here.*

## Pull Request Checklist

Please review and check all that apply before submitting your PR:

- [ ] The code compiles and passes all tests
- [ ] All new and existing tests are passing
- [ ] Documentation has been updated (if applicable)
- [ ] Code follows project style guidelines
- [ ] No sensitive information is included
- [ ] Linked relevant issues (if any)
- [ ] Added necessary reviewers

### CODE REVIEW

**Review by reh-sfl — CHANGES_REQUESTED** [review]
Like we did for the Review tab, it would be good practice to add metadata to the JSON output:

```json
{
  "version": 1,
  "exported_at": "2026-07-22T13:34:07.396943+00:00",
  ...
}
```

Also, we shouldn't export attributes with a null value.

**Review by ValentinBoudevinSFL — CHANGES_REQUESTED** [review]
Please add CLI commands for this feature. 

Can be useful to import/export via command line.

**Review by reh-sfl — DISMISSED** [review]

**Review by ValentinBoudevinSFL — DISMISSED** [review]
1- Image

Structure seems redundant for the project with each variant containing all projects details. 

Instead of: 


```
[
  {
    "codebase_path": null,
    "description": "test project description",
    "environment": null,
    "other_info": null,
    "project_name": "test ai context import/export",
    "risks": null,
    "threat_model": "test threat model",
    "variant_description": null,
    "variant_name": "test"
  }
]
```

Someting like the following seems more appropriate: 

```
{
 "project_name": "test ai context import/export",
 "project_description": "test project description",
  "variants" : 
      [
        {
          "variant_name": "test"
          "variant_description": null,
          "codebase_path": null,
          "environment": null,
          "other_info": null,
          "risks": null,
          "threat_model": "test threat model",
        }
      ]
``` 

2- 

Image

I don't think this button is necessary in addition to the top one to export

File: frontend/src/pages/AIContext.tsx [hunk 1]
```diff
@@ -415,52 +670,18 @@ function AIContext() {
...
-                                    onClick={() => handleDeleteFile(f.id)}
-                                    className="ml-auto text-zinc-400 hover:text-red-400"
-                                >
-                                    <FontAwesomeIcon icon={faXmark} />
-                                </button>
-                            </li>
-                        ))}
-                    </ul>
-                )}
+                {/* Export this variant */}
```
Comment (sfl-pe-sync) [inline frontend/src/pages/AIContext.tsx]: This removes the existing supplemental-file upload, listing, and deletion functionality even though import/export does not replace it. Preserve those controls and handlers alongside the new context export button unless file support is being intentionally deprecated with a documented migration.

File: frontend/src/pages/AIContext.tsx [hunk 2]
```diff
@@ -194,37 +247,105 @@ function AIContext() {
...
+    };
+
+    const projectNameById = useCallback(
+        (projectId: string) => projects.find(p => p.id === projectId)?.name ?? '',
+        [projects]
+    );
+
+    const handleExportSelected = async () => {
+        if (ioBusy || exportSelection.size === 0) return;
+        setIoBusy(true);
```
Comment (sfl-pe-sync) [inline frontend/src/pages/AIContext.tsx:264]: Exporting selected variants downloads every project's context and filters it client-side. This unnecessarily exposes potentially sensitive context to the browser and scales poorly. Send the selected variant IDs to a filtered backend export endpoint, or request each selected variant directly.

**Review by sfl-pe-sync — CHANGES_REQUESTED** [review]
The import/export workflow is generally structured well, but it introduces an unrelated feature regression and retrieves substantially more sensitive data than requested.

File: src/bin/cmd_context.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    imported = result["imported"]
+    ignored = result["ignored"]
+    failed = result["failed"]
+
+    for item in ignored:
+        click.echo(f"  Ignored {_label(item)}: {item.get('reason', '')}", err=True)
+    for item in failed:
+        click.echo(f"  Failed {_label(item)}: {item.get('reason', '')}", err=True)
+
```
Comment (sfl-pe-sync) [inline src/bin/cmd_context.py:99]: The command exits successfully even when `failed` contains rejected entries. Consequently, `src/entrypoint.sh --import-context` and CI pipelines report success after an incomplete import. Exit non-zero when any entry fails, while still reporting the partial results, and add a CLI regression test asserting the exit code.

File: frontend/src/pages/AIContext.tsx [hunk 4]
```diff
@@ -82,15 +123,28 @@ function AIContext() {
      
     }, []);
 
+    // Load all variants for the export selector (lazily, when the menu first opens)
+    const [allVariantsLoaded, setAllVariantsLoaded] = useState(false);
+    const loadAllVariants = useCallback(() => {
+        if (allVariantsLoaded) return;
+        Variants.listAll().then(vs => {
```
Comment (sfl-pe-sync) [inline frontend/src/pages/AIContext.tsx:130]: `Variants.listAll()` uses a handler that swallows HTTP errors, so a failed request can resolve as an empty list. This then sets `allVariantsLoaded` to true, displays no variants, and prevents reopening the menu from retrying. Use a throwing fetch helper or make `listAll` reject non-OK responses before marking the data loaded.

**Review by sfl-pe-sync — CHANGES_REQUESTED** [review]
The context import/export implementation is not ready to merge: partial CLI imports can falsely report success, and export-selector loading can silently convert API failures into a permanently empty list. Review was limited to the supplied diff; application startup and final-container execution were not performed.

**Review by sfl-pe-sync — DISMISSED** [review]
The context import/export flow has a blocking round-trip contract defect: valid existing context can be exported but rejected when imported. The PR should not merge until export and import use compatible validation semantics.

---
**Additional comments (lines outside diff hunk):**
- `src/helpers/context_io.py` [error]: Exported documents are not guaranteed to be importable: the export schema permits null variant descriptions and threat models, matching existing nullable context fields, while import treats description and threat model as mandatory. A normal export containing incomplete context will therefore fail its own round trip. Preserve nullable exported values during import, or exclude incomplete entries consistently and document that behavior; add a regression test exporting and re-importing context with null fields.

**Review by reh-sfl — APPROVED** [review]

**Review by ValentinBoudevin — APPROVED** [review]
